src/components/checkbox/Checkbox.js

```diff
@@ -243,4 +243,12 @@ export default class CheckBoxComponent extends Field {
 
     return changed;
   }
+
+  calculateComponentValue(data, flags, row) {
+    if (this.options.pdf && _.get(this.root, 'submission._id', false)) {
+      flags = flags || {};
+      flags.fromSubmission = true;
+    }
```

Comment:
I do not really like the idea to set fromSubmission flag somewhere outside setSubmission logic. Maybe, we can set  calculationLocked to true instead or something else. Why are there no calculation issues with other form components? What is special in checkbox functionality that causes this issue?